extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClientRecorder.java

```diff
@@ -189,18 +200,19 @@ private AuthenticationMechanism getAuthenticationMechanism(String authMechanism)
         return mechanism;
     }
 
-    private MongoCredential createMongoCredential(CredentialConfig config) {
-        String username = config.username.orElse(null);
+    private MongoCredential createMongoCredential(MongoClientConfig config) {
+        String username = config.credentials.username.orElse(null);
         if (username == null) {
             return null;
         }
 
-        char[] password = config.password.map(String::toCharArray).orElse(null);
-        //admin is the default auth source in mongo and null is not allowed
-        String authSource = config.authSource.orElse("admin");
+        char[] password = config.credentials.password.map(String::toCharArray).orElse(null);
+        // get the authsource, or the database from the config, or 'admin' as it is the default auth source in mongo
+        // and null is not allowed
+        String authSource = config.credentials.authSource.orElse(config.database.orElse("admin"));
```

Comment:
Well, actuall the docs about `authsource` say:

> The value defaults to the database specified in the path portion of the connection string

But it's a lie, because I don't see that being used here. Not your fault though, but still looks like a bug to me.

Reply:
IIt was in the original code, I keep it this way but maybe this should be revisited. 
In fact it's a little complex as if you set credential inforamtion in the URL, it's true, but if you set credential information in the CredentialConfig properties an algorithm is used that mimics what the behaviour with URL does ...

So this should be rephrased by
```
This value default to the one specified in the admin property or admin if not set.
``` 

The guide clearly states that everything can be added to the URL and this is the more easier way to configure the connection to MongoDB.